### ISSUE TITLE
New React fragments syntax causes compilation error

### ISSUE THREAD

**Issue by papodaca** [jashkenas/coffeescript#4802]
Using the new fragment syntax supported by react [16.2](https://reactjs.org/blog/2017/11/28/react-v16.2.0-fragment-support.html) causes the following error:

```
ERROR in ./app/javascript/components/layout/index.coffee
Module build failed: Error: ./app/javascript/components/layout/index.coffee:10:4: error: missing / (unclosed regex)
  </>
   ^
    L9:   </>
```

Example input:
```coffeescript
import React from 'react'

Component = (props) =>
  <>
    <OtherComponent />
    <OtherComponent />
  </>

export default Component
```

Expected output:
```javascript
var Component;

import React from 'react';

Component = (props) => {
  return <>
    <OtherComponent />
    <OtherComponent />
  </>;
};

export default Component;
```

**Comment by GeoffreyBooth** [comment]
Well, at least it’s a compile error at the moment, rather than compiling into something else. As a workaround for now, follow the [advice](https://reactjs.org/blog/2017/11/28/react-v16.2.0-fragment-support.html#other-tools) of Facebook’s announcement:

> For other tools, please check with the corresponding documentation to check if there is support available. However, if you’re blocked by your tooling, you can always start with using the `<Fragment>` component and perform a codemod later to replace it with the shorthand syntax when the appropriate support is available.

```jsx
Component = (props) =>
  <Fragment>
    <OtherComponent />
    <OtherComponent />
  </Fragment>
```

**Comment by GeoffreyBooth** [comment]
@clemmy I know we’re not as cool as Babel, Flow and TypeScript, but we’d appreciate being added to the list of tools that support JSX that you think about when making changes to it 😄

**Comment by clemmy** [comment]
@GeoffreyBooth For sure 🙂. You can keep up to date on our plans/proposals for JSX [here](https://github.com/facebook/jsx/). If you can reference the PR/shoot a comment in there to notify us of the CoffeeScript progress, we'd be happy to add you into the list!

**Comment by GeoffreyBooth** [comment]
Fixed by #4804.
